### PR TITLE
Updated Core Tools pipeline to enable building for Integration Tests

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
This PR contains the following:

* Updated Core Tools pipeline (yml file) to enable building for Integration Tests
* Renamed from `integrationTestBuildManifest.json` to `integrationTestsBuildManifest.json` to keep language consistency between repos
* Moved `uploadContentToStorageAccount.ps1` script to repo root and added `-AllowClobber` to the `Install-Module Az.Storage` command
* Disabled running the Core Tools E2E tests when building for Integration Tests
* Removed `'no-runtime'` from the list of target runtimes to be built

### CODE REVIEW

**Comment by Francisco-Gamino** [comment]
Hi @AnatoliB -- Could you please take a look? I'd like to get your input on the changes for `FilterPowershellRuntimes()`.

File: build/BuildSteps.cs [hunk 1]
```diff
@@ -157,18 +155,17 @@ public static void FilterPowershellRuntimes()
                     var powerShellVersion = Path.GetFileName(powershellWorkerPath);
                     var powershellRuntimePath = Path.Combine(powershellWorkerPath, "runtimes");
 
-                    var allKnownPowershellRuntimes = Settings.ToolsRuntimeToPowershellRuntimes[powerShellVersion].Values.SelectMany(x => x).Distinct().ToList();
                     var allFoundPowershellRuntimes = Directory.GetDirectories(powershellRuntimePath).Select(Path.GetFileName).ToList();
+                    var powershellForCurrentRuntime = Settings.ToolsRuntimeToPowershellRuntimes[powerShellVersion][runtime];
```
Comment (AnatoliB) [inline build/BuildSteps.cs:159]: I'm having some troubles understanding the `powershellForCurrentRuntime` variable name. Shouldn't this be something like `powershellRuntimesForCurrentToolsRuntime`?

Comment (Francisco-Gamino) [reply build/BuildSteps.cs:159]: Yes, it should. I will update it.

**Review by AnatoliB — APPROVED** [review]
Signing-off for the PowerShell part

**Comment by Francisco-Gamino** [comment]
> Signing-off for the PowerShell part

Thanks @AnatoliB !

**Review by soninaren — APPROVED** [review]

**Pull request merged**